### PR TITLE
Feature/export playlist spotify

### PR DESCRIPTION
**Pull request by leilahamm**
Features and Updates:
- Integration of Spotify API ViewModel in Playlist Overview Screen:
  - Added the SpotifyApiViewModel to the PlaylistOverviewScreen to handle playlist export functionality.
- Export Playlist Logic:
  - Implemented the export functionality in the PlaylistOverviewScreen, connected to the "Export" button.
  - When the "Export" button is clicked:
    - A confirmation dialog is displayed, asking the user to confirm the action.
    - Upon confirmation:
      - The playlist is exported to Spotify using the SpotifyApiViewModel.
      - If the export is successful, the playlist is deleted from the app's database.
      - A success toast is displayed, and the user is redirected to the Library screen.
      - If the export fails, an error toast is displayed.
- Creating a List of Spotify Tracks in Playlist ViewModel:
  - Added a function in the PlaylistViewModel to generate a list of SpotifyTrack objects for the 50 most liked songs in the playlist.
  - This function is used during the export process to ensure only the top-liked tracks are added to the exported playlist.
- Enhancements to createBeatLinkPlaylist in Spotify API ViewModel:
  - Updated the createBeatLinkPlaylist function to include an onResult callback.
  - The onResult callback returns the ID of the playlist created in Spotify, enabling better handling of success or failure scenarios during the export process.
- Update navigation to handle the stack when exporting a playlist
- Remove hardcoded value for consistent code

### CODE REVIEW

File: app/src/main/java/com/epfl/beatlink/ui/library/PlaylistOverview.kt [hunk 1]
```diff
@@ -209,9 +229,70 @@ fun PlaylistOverviewScreen(
...
+                          Log.d("Spotify", idSpotify.toString())
+                          if (idSpotify != null) {
+                            // Delete playlist only if creation was successful
+                            playlistViewModel.deletePlaylist(selectedPlaylistState.playlistID)
+                            showDialogExport = false
+                            Toast.makeText(
+                                    context, "Playlist exported successfully", Toast.LENGTH_SHORT)
+                                .show()
+                            navigationActions.navigateTo(Screen.LIBRARY)
+                          } else {
```
Comment (lunasrc) [inline app/src/main/java/com/epfl/beatlink/ui/library/PlaylistOverview.kt:274]: I think you should use the `navigateToAndClearBackStack(screen: String, entriesToClear: Int)` function to block the navigation to the `PlaylistOverviewScreen` of the playlist that just got exported when navigating back via the phone back button. Let me know if the current code works or it needs to be changed.

File: app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt [hunk 2]
```diff
@@ -184,6 +185,19 @@ class PlaylistViewModel(
...
 
+  /**
+   * Create the final list of track to export the playlist that contain the 50 most liked songs in
+   * descending order
+   */
+  fun getFinalListTracks(): List<SpotifyTrack> {
+    return selectedPlaylist_.value
+        ?.playlistTracks
+        ?.sortedByDescending { it.likes } // Sort by likes in descending order
+        ?.take(50) // Take at most 50 tracks
```
Comment (lunasrc) [inline app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt:196]: I would suggest to add the number of tracks as an optional parameter: fun `getFinalListTracks(maxTracks: Int = 50): List<SpotifyTrack>` or to define it as a constant: `private const val DEFAULT_TRACK_LIMIT = 50` (in `Playlist.kt`)

**Review by lunasrc — CHANGES_REQUESTED** [review]
The implementation is solid and well-written, with clear logic and structure that made it easy to understand. The PR meets its objectives effectively and is close to approval.

File: app/src/main/java/com/epfl/beatlink/ui/library/PlaylistOverview.kt (hunk 1 above)
Comment (leilahamm) [reply app/src/main/java/com/epfl/beatlink/ui/library/PlaylistOverview.kt:274]: You are absolutely right the back button was returning to the overview of the deleted playlist I will fix that. Thanks

**Review by lunasrc — APPROVED** [review]
The implementation in this PR is clear, well-structured, and effectively achieves the intended functionality. Approved!

**Pull request merged**